Mathlib/RingTheory/Idempotents.lean

```diff
@@ -111,6 +112,15 @@ structure OrthogonalIdempotents : Prop where
   idem : ∀ i, IsIdempotentElem (e i)
   ortho : Pairwise (e · * e · = 0)
 
+/-- A primitive idempotent is primitive if it's not zero and `e ≠ e₁ + e₂` for any
+  `e₁` and `e₂` idempotents and pairwisely orthogonal. -/
+structure IsPrimitiveIdempotents (e : R): Prop where
+  idem : IsIdempotentElem e
+  [ne_zero : e ≠ 0]
+  ne_sum_ortho' :
+    ∀ (f g : R) (_ : OrthogonalIdempotents ![f, g])
+      (_ : f ≠ 0) (_ : g ≠ 0), e ≠ f + g
```

Comment:
Is it better to use `[NeZero f]` and `[NeZero g]` here? Not too sure

Reply:
should be the same though

Reply:
done